Update temperature reading docs (#333)
icub-tech-iit/documentation: docs/temperature_sensors/general/overview.md

## Motor temperature Sensors

The Motor Temperature Sensors are simple sensors made of a specific type of resistor (PT100 or PT1000) and of a small board defined as the **Temperature Detection Board (TDB)**, which has the goal of decoding the value read by the resistor giving out a raw output code, which can be then translated to Celsius Degree so that it can be easily understood by the end-user.

> In a nutshell, most of the `ergoCub` motors are provided with an internal resistance temperature detector, which is then connected to a TDB board. This one is then linked to a `2FOC` board, which elaborates that data coming from the `I2C` and sends them via `CAN` to an `EMS` board. Finally, this one streams the raw temperature values via `ETH` to `embObjMotionControl` which updates and stores all of these data, ready to be requested by a user external module.

Reviewer: ```suggestion
In a nutshell, most of the `ergoCub` motors are provided with an internal resistance temperature detector, which is then connected to a TDB board. The TDB board is then linked to a `2FOC` board, which elaborates the data acquired via `I2C` bus and sends it via the `CAN` bus to an `EMS` board. Finally, the EMS board streams the raw temperature values via the `ETH` bus to `embObjMotionControl`, which in turn updates and stores these acquisitions that the user can readily request from an external module.
```

That being done, it should be possible to use the features of the temperature sensors.

### How to read the motor temperature

> First of all, the user should be sure that the sensor has been mounted on the desired motors, and then he/she needs to update the robot configuration files accordingly, as described in the [configuration section](#how-to-update-the-configuration-files). 

Reviewer: ```suggestion
First off, the user ought to make sure that the sensor is mounted on the desired motors, and then needs to update the robot configuration files accordingly, as described in the [configuration section](#how-to-update-the-configuration-files). 
```

> The correct configuration enables the motor temperature reading and sending the values to the yarprobotinterface, but, currently, these measures are not published on the yarp `stateExt` port. Therefore the user needs to develop a module application similar to [this one](https://github.com/MSECode/motionControlModule), to read the temperature.

Reviewer: ```suggestion
The correct configuration allows reading and sending the motor temperature values to the `yarprobotinterface`. However, these measurements are not currently published on the YARP `stateExt` port. Therefore, the user needs to devise a module application, similar to [this one](https://github.com/MSECode/motionControlModule), to read the temperature.
```

In addition, the user module can retrieve the `warningTemperatureLimits` and check if the temperature exceeds that limit to put the motor in a safer state with lower current consumption.

> Is it important to notice that the firmware puts the motor in fault when the temperature goes beyond the `hardwareTemperatureLimits` for longer than 10 seconds.

Reviewer: ```suggestion
!!! warning "HW Fault due to overheating"

    Is it important to notice that the firmware puts the motor in HW Fault when the temperature 
    goes beyond the `hardwareTemperatureLimits` for longer than 10 seconds.
```
